src/dvc_data/hashfile/meta.py

```diff
@@ -30,6 +31,8 @@ class Meta:
     inode: Optional[int] = None
     mtime: Optional[float] = None
 
+    remote: Optional[str] = field(default=None, eq=False)
```

Comment:
This feels like the wrong abstraction to me, `remote` name has nothing to do with fs level file metadata. I'd still prefer having something like `Output.meta` and `DataIndexEntry.meta` being changed to dicts mapping `remote -> remote_specific_meta`. IMO we should really only be setting individual meta fields via `Meta.from_info(fs.info(...))` and we shouldn't be mutating fields after the `Meta` instance has been created.

But I guess since we aren't supporting multiple remotes here maybe this is good enough for now.

Reply:
I don't like this either. I am open to any suggestions. Ideally, there should be `meta` for each of the remotes. But this means that there can't be a `repo.index` (i.e. repo level index).

Also, since we use `Output.get_obj()` which returns a `Tree/HashFile` (which is where the metadata for `files` come from), I am not sure of a good way to incorporate this in a simple way.

One way might be to set this in `Output.last_pushed_remote_name` or something, and mutate it. But that is not tied to `Output.get_obj()` or `Tree/HashFile` from where we dump. Let me know what you think.

Reply:
Maybe this means we should consider dropping `output.files` as it currently exists in favor of having proper exports. Then `out.meta` would always the the repo/local meta, `repo.index` would always be the repo level index, and we would then add separate export/remote index (w/its own separate remote meta). Not sure what that would look like in serialization though, and this probably goes beyond the scope of what we can do in the short term for cloud versioning.

Reply:
yeah, a correct way would require us to do all that, which is required by multi-remote support. Also, an index or something similar that is multi-remote (don't have to be dataindex) is also required so that we can unify the checkout/pull.